<?xml version="1.0" encoding="UTF-8"?>

<launch>
  <arg name="play_bag" default="true" />
  <arg name="bag_file" default="/PATH/TO/basement_dataset.bag"/>

  <node name="voxblox_node" pkg="voxblox_ros" type="tsdf_server" output="screen" clear_params="true">
    <param name="method" value="fast" />
    <param name="verbose" value="false" />
    <param name="color_mode" value="color" />
    <param name="enable_icp" value="false" />
    <param name="mesh_min_weight" value="2" />
    <param name="world_frame" value="world" />
    <param name="tsdf_voxel_size" value="0.05" />
    <param name="max_ray_length_m" value="3.0" />
    <param name="use_const_weight" value="true" />
    <param name="truncation_distance" value="0.5" />
    <param name="tsdf_voxels_per_side" value="16" />
    <param name="voxel_carving_enabled" value="true" />
    <param name="icp_refine_roll_pitch" value="false" />
    <param name="update_mesh_every_n_sec" value="1.0" />
    <remap from="pointcloud" to="/camera/depth/points"/>
    <param name="mesh_filename" value="$(find voxblox_ros)/mesh_results/$(anon basement).ply" />
  </node>
</launch>